src/utils/exchange.js

```diff
@@ -95,3 +105,46 @@ export const parseOffer = (
 export const isWethConvertedTx = (fromAssetSymbol: string, contractAddress: string): boolean => {
   return fromAssetSymbol === ETH && contractAddress === WETH[chainId].address;
 };
+
+/* eslint-disable i18next/no-literal-string */
+const setAllowanceAbiFunction = [{
+  name: 'approve',
+  outputs: [{ type: 'bool', name: 'out' }],
+  inputs: [{ type: 'address', name: '_spender' }, { type: 'uint256', name: '_value' }],
+  constant: false,
+  payable: false,
+  type: 'function',
+  gas: 38769,
+}];
+
+const ethProvider = () => getEthereumProvider(getEnv().NETWORK_PROVIDER);
+
+export const createAllowanceTx = async (
+  fromAssetAddress: string,
+  clientAddress: string,
+  contractAddress: string,
+): Promise<AllowanceTransaction | null> => {
+  if (!clientAddress) {
+    reportOrWarn('Unable to set allowance', null, 'error');
+    return null;
+  }
+  try {
+    const encodedContractFunction = encodeContractMethod(
+      setAllowanceAbiFunction,
+      'approve',
+      [contractAddress, constants.MaxUint256.toString()],
+    );
+
+    const txCount = await ethProvider().getTransactionCount(clientAddress);
+
+    return {
+      nonce: txCount.toString(),
+      to: fromAssetAddress,
+      chainId: '1',
+      data: encodedContractFunction,
+    };
+  } catch (e) {
```

Comment:
I'm suspecting that `encodeContractMethod` is the reason for the `try / catch`es here

Reply:
And `ethProvider().getTransactionCount()`